packages/gateway/src/graphql/config.ts

```diff
@@ -105,7 +105,10 @@ export const getApolloConfig = (): Config => {
         throw new AuthenticationError(err)
       }
 
-      return { Authorization: request.headers.authorization }
+      return {
+        Authorization: request.headers.authorization,
+        'x-correlation-id': request.headers['x-correlation-id'] || uniqueId()
+      }
```

Comment:
This will ensure that all the resolver requests for a particular GQL query and mutations(which we can consider as our transaction) will have the same `x-correlation-id` right?

Reply:
Precisely! As our client doesn't currently send correlation ids (like it should in the future), this will create one right after the gateway receives the request. This id will then flow through all services and openhim and even remain the same if at some point the request chain comes back to gateway (not sure if that can happen in reality)

Reply:
Nice! This will definitely make tracking the requests and responses much easier.